src/traditional_de/automated.jl

```diff
@@ -51,28 +52,29 @@ w::Int=1, rtol=10.0, atol=2.0, τs = 1:100, metric = Euclidean()
 [^Hegger1999]: Hegger & Kantz, [Improved false nearest neighbor method to detect determinism in time series data. Physical Review E 60, 4970](https://doi.org/10.1103/PhysRevE.60.4970).
 """
 function optimal_traditional_de(s::AbstractVector, delaymethod::String= "mi_min",
-    dimensionmethod::String = "afnn"; thres::Real = 0.05, dmax::Int = 10,
-    w::Int=1, rtol=10.0, atol=2.0, τs = 1:100, metric = Euclidean())
+        dimensionmethod::String = "afnn";
+        thres::Real = 0.05, dmax::Int = 10, w::Int=1, rtol=10.0, atol=2.0, τs = 1:100,
+        metric = Euclidean(), r=2.0, w=1
+    )
 
     @assert dimensionmethod ∈ ("afnn", "fnn", "ifnn", "f1nn")
-
     τ = estimate_delay(s, delaymethod, τs)
-    γs = 1:dmax-1 # TODO: This must be updated to just dimension only
+    γs = 1:dmax-1 # TODO: This must be updated to dimension in 2.0
 
     if dimensionmethod=="afnn"
-        dimension_statistic = estimate_dimension(s, τ, γs, dimensionmethod)
+        dimension_statistic = afnn(s, τ, γs, metric)
         Y, τ = cao_embed(s, τ, dimension_statistic, thres)
         E2 = stochastic_indicator(s, τ, γs)
         flag = is_stochastic(E2, thres)
```

Comment:
This one is still based on the assumption that E2 converges to 1, which I believe is not always the case. I will fix that by replacing it with the slope-convergence as you have suggested it.